UnitTests/DimTests.cs

```diff
@@ -317,6 +317,184 @@ public void Dim_Validation_Do_Not_Throws_If_NewValue_Is_DimAbsolute_And_OldValue
+		public void Only_DimAbsolute_And_DimFactor_As_A_Different_Procedure_For_Assigning_Value_To_Width_Or_Height ()
+		{
+			// Testing with the Button because it properly handles the Dim class.
+
+			Application.Init (new FakeDriver (), new NetMainLoop (() => FakeConsole.ReadKey (true)));
+
+			var t = Application.Top;
+
+			var w = new Window ("w") {
+				Width = 100,
+				Height = 100
+			};
+
+			var f1 = new FrameView ("f1") {
+				X = 0,
+				Y = 0,
+				Width = Dim.Percent (50),
+				Height = 5
+			};
+
+			var f2 = new FrameView ("f2") {
+				X = Pos.Right (f1),
+				Y = 0,
+				Width = Dim.Fill (),
+				Height = 5
+			};
+
+			var v1 = new Button ("v1") {
+				X = Pos.X (f1) + 2,
+				Y = Pos.Bottom (f1) + 2,
+				Width = Dim.Width (f1) - 2,
+				Height = Dim.Fill () - 2
+			};
+
+			var v2 = new Button ("v2") {
+				X = Pos.X (f2) + 2,
+				Y = Pos.Bottom (f2) + 2,
+				Width = Dim.Width (f2) - 2,
+				Height = Dim.Fill () - 2
+			};
+
+			var v3 = new Button ("v3") {
+				Width = Dim.Percent (10),
+				Height = Dim.Percent (10)
+			};
+
+			var v4 = new Button ("v4") {
+				Width = Dim.Sized (50),
+				Height = Dim.Sized (50)
+			};
+
+			var v5 = new Button ("v5") {
+				Width = Dim.Width (v1) - Dim.Width (v3),
+				Height = Dim.Height (v1) - Dim.Height (v3)
+			};
+
+			var v6 = new Button ("v6") {
+				X = Pos.X (f2),
+				Y = Pos.Bottom (f2) + 2,
+				Width = Dim.Percent (20, true),
+				Height = Dim.Percent (20, true)
+			};
+
+			w.Add (f1, f2, v1, v2, v3, v4, v5, v6);
+			t.Add (w);
+
+			t.Ready += () => {
+				Assert.Equal ("Dim.Absolute(100)", w.Width.ToString ());
+				Assert.Equal ("Dim.Absolute(100)", w.Height.ToString ());
+				Assert.Equal (100, w.Frame.Width);
+				Assert.Equal (100, w.Frame.Height);
+
+				Assert.Equal ("Dim.Factor(factor=0,5, remaining=False)", f1.Width.ToString ());
+				Assert.Equal ("Dim.Absolute(5)", f1.Height.ToString ());
+				Assert.Equal (49, f1.Frame.Width); // 50-1=49
+				Assert.Equal (5, f1.Frame.Height);
+
+				Assert.Equal ("Dim.Fill(margin=0)", f2.Width.ToString ());
+				Assert.Equal ("Dim.Absolute(5)", f2.Height.ToString ());
+				Assert.Equal (49, f2.Frame.Width); // 50-1=49
+				Assert.Equal (5, f2.Frame.Height);
+
+				Assert.Equal ("Dim.Combine(DimView(side=Width, target=FrameView()({X=0,Y=0,Width=49,Height=5}))-Dim.Absolute(2))", v1.Width.ToString ());
+				Assert.Equal ("Dim.Combine(Dim.Fill(margin=0)-Dim.Absolute(2))", v1.Height.ToString ());
+				Assert.Equal (47, v1.Frame.Width); // 49-2=47
+				Assert.Equal (89, v1.Frame.Height); // 98-5-2-2=89
+
+
+				Assert.Equal ("Dim.Combine(DimView(side=Width, target=FrameView()({X=49,Y=0,Width=49,Height=5}))-Dim.Absolute(2))", v2.Width.ToString ());
+				Assert.Equal ("Dim.Combine(Dim.Fill(margin=0)-Dim.Absolute(2))", v2.Height.ToString ());
+				Assert.Equal (47, v2.Frame.Width); // 49-2=47
+				Assert.Equal (89, v2.Frame.Height); // 98-5-2-2=89
+
+				Assert.Equal ("Dim.Factor(factor=0,1, remaining=False)", v3.Width.ToString ());
```

Comment:
You must specify a format specifier here to ensure that a period `.` is used for decimals. For the tests to run reliably in both local environments and in CI/CD the way #s are formatted has to be explicit. You used `,` instead of `.`.

See Line 170.

`Assert.Equal ($"Dim.Factor(factor={f / 100:0.###}, remaining={false})", dim.ToString ());`

Reply:
I should get that error on my pc. That bad. I will fix that. Thanks

Reply:
This change throws exceptions in all cultures and not just in github.